crates/services/block_aggregator_api/src/api/protobuf_adapter.rs

```diff
@@ -47,34 +66,35 @@ impl Server {
 
 #[async_trait]
 impl BlockAggregator for Server {
-    async fn get_block_height(
+    async fn get_synced_block_height(
         &self,
         request: tonic::Request<ProtoBlockHeightRequest>,
     ) -> Result<tonic::Response<ProtoBlockHeightResponse>, tonic::Status> {
         tracing::debug!("get_block_height: {:?}", request);
+        tracing::info!("get_block_height: {:?}", request);
         let (response, receiver) = tokio::sync::oneshot::channel();
         let query = BlockAggregatorQuery::GetCurrentHeight { response };
         self.query_sender.send(query).await.map_err(|e| {
             tonic::Status::internal(format!("Failed to send query: {}", e))
         })?;
         let res = receiver.await;
+        tracing::info!("query result: {:?}", &res);
         match res {
             Ok(height) => Ok(tonic::Response::new(ProtoBlockHeightResponse {
-                height: *height,
+                height: height.map(|inner| *inner),
             })),
             Err(e) => Err(tonic::Status::internal(format!(
                 "Failed to receive height: {}",
                 e
             ))),
         }
     }
-    type GetBlockRangeStream = ReceiverStream<Result<ProtoBlockResponse, Status>>;
+    type GetBlockRangeStream = BoxStream<Result<ProtoBlockResponse, Status>>;
 
     async fn get_block_range(
         &self,
         request: tonic::Request<ProtoBlockRangeRequest>,
     ) -> Result<tonic::Response<Self::GetBlockRangeStream>, tonic::Status> {
-        const ARB_LITERAL_BLOCK_BUFFER_SIZE: usize = 100;
         let req = request.into_inner();
         let (response, receiver) = tokio::sync::oneshot::channel();
         let query = BlockAggregatorQuery::GetBlockRange {
```

Comment:
Looks like each query will go to main `run` loop, so all read queries will become sequential, instead of parallel. Looks like you could pass database/s3 configuraiton here, and directly fetch the data fro mthe soruce, isntead of sending a query to the service, which will fetch it from the source for you.